packages/libp2p/src/transport-manager.ts

```diff
@@ -29,7 +29,8 @@ export interface DefaultTransportManagerComponents {
 export class DefaultTransportManager implements TransportManager, Startable {
   private readonly components: DefaultTransportManagerComponents
   private readonly transports: Map<string, Transport>
-  private readonly listeners: Map<string, Listener[]>
+  /** Map<TransportKey, Map<Multiaddr Listener>> */
+  private readonly listeners: Map<string, Map<Multiaddr, Listener>>
```

Comment:
I may be missing something, but this change seems to have been made just to support the `removeListener` method, itself only added to be invoked in a test.

Can this be reverted?